test_util/cluster.py

```diff
@@ -184,6 +202,86 @@ def check_ssh(self, hosts):
         """
         self.ssher.remote_cmd(hosts, ['echo'])
 
+    def mesos_metrics_snapshot(self, host):
+        """Return a snapshot of the Mesos metrics for host."""
+        if host in self.masters:
+            port = 5050
+        else:
+            port = 5051
+
+        with self.ssher.tunnel(host) as tunnel:
+            return json.loads(
+                tunnel.remote_cmd(
+                    curl_cmd + ['{}:{}/metrics/snapshot'.format(host.private_ip, port)]
+                ).decode('utf-8')
+            )
+
+    def zk_mode(self, host):
+        """Return the mode of the ZooKeeper instance on host."""
+        with self.ssher.tunnel(host) as tunnel:
+            stat_out = tunnel.remote_cmd([
+                'echo', 'stat', '|', '/opt/mesosphere/bin/toybox', 'nc', 'localhost', '2181'
+            ])
+        for line in stat_out.split(b'\n'):
+            line = line.strip()
+            if line.startswith(b'Mode: '):
+                mode = line.split(b':')[1].strip()
+                if mode not in (b'leader', b'follower', b'standalone'):
```

Comment:
It is, but implicitly. This is a more direct assertion that'll fail earlier and with a clearer message. I wanted this function to be responsible for retrieving (and validating) the mode of a single host, and `master_zk_modes()` to be responsible for retrieving (and validating) the modes of the entire cluster. Host IPs are included in the error message in both places.